src/Symfony/Component/Mailer/Bridge/Sweego/RemoteEvent/SweegoPayloadConverter.php

```diff
@@ -13,20 +13,32 @@
 
 use Symfony\Component\RemoteEvent\Event\Mailer\AbstractMailerEvent;
 use Symfony\Component\RemoteEvent\Event\Mailer\MailerDeliveryEvent;
+use Symfony\Component\RemoteEvent\Event\Mailer\MailerEngagementEvent;
 use Symfony\Component\RemoteEvent\Exception\ParseException;
 use Symfony\Component\RemoteEvent\PayloadConverterInterface;
 
 final class SweegoPayloadConverter implements PayloadConverterInterface
 {
     public function convert(array $payload): AbstractMailerEvent
     {
-        if (\in_array($payload['event_type'], ['email_sent', 'delivered'], true)) {
+        if (\in_array($payload['event_type'], ['email_sent', 'delivered', 'soft-bounce', 'hard_bounce'], true)) {
             $name = match ($payload['event_type']) {
                 'email_sent' => MailerDeliveryEvent::RECEIVED,
                 'delivered' => MailerDeliveryEvent::DELIVERED,
+                'soft-bounce', 'hard_bounce' => MailerDeliveryEvent::BOUNCE,
```

Comment:
Off topic, but the delivered and received events are past tense, shouldn't it be bounce**D**?

Reply:
Same for Engagement events (`open` -> `opened`, `click` -> `clicked`, `unsubscribe` -> `unsubscribed`), I guess?

Reply:
The MailerDeliveryEvent constants are the symfony/mailer abstraction. They are not specific to Sweego, and so the name might not match (other providers have different names for their event types anyway)